src/crypto/ecies.ts

```diff
@@ -0,0 +1,265 @@
+    assert(msg.length <= 32, 'Message is too long')
+    resolve(Buffer.from(ec.sign(msg, privateKey, { canonical: true }).toDER()))
+  })
+}
+
+export function verify(publicKey: Buffer, msg: Buffer, sig: Buffer) {
+  return new Promise(function (resolve, reject) {
+    assert(publicKey.length === 65 || publicKey.length === 33, 'Bad public key')
+    if (publicKey.length === 65) {
+      assert(publicKey[0] === 4, 'Bad public key')
+    }
+    if (publicKey.length === 33) {
+      assert(publicKey[0] === 2 || publicKey[0] === 3, 'Bad public key')
+    }
+    assert(msg.length > 0, 'Message should not be empty')
+    assert(msg.length <= 32, 'Message is too long')
+    if (ec.verify(msg, sig, publicKey)) {
+      resolve(null)
+    } else {
+      reject(new Error('Bad signature'))
+    }
+  })
+}
+
+export function derive(
+  privateKeyA: Buffer,
+  publicKeyB: Buffer
+): Promise<Buffer> {
+  return new Promise(function (resolve) {
+    assert(Buffer.isBuffer(privateKeyA), 'Bad private key')
+    assert(Buffer.isBuffer(publicKeyB), 'Bad public key')
+    assert(privateKeyA.length === 32, 'Bad private key')
+    assert(isValidPrivateKey(privateKeyA), 'Bad private key')
+    assert(
+      publicKeyB.length === 65 || publicKeyB.length === 33,
+      'Bad public key'
+    )
+    if (publicKeyB.length === 65) {
+      assert(publicKeyB[0] === 4, 'Bad public key')
+    }
+    if (publicKeyB.length === 33) {
+      assert(publicKeyB[0] === 2 || publicKeyB[0] === 3, 'Bad public key')
+    }
+    const keyA = ec.keyFromPrivate(privateKeyA)
+    const keyB = ec.keyFromPublic(publicKeyB)
+    const Px = keyA.derive(keyB.getPublic()) // BN instance
+    resolve(Buffer.from(Px.toArray()))
+  })
+}
+
+export async function encrypt(
+  publicKeyTo: Buffer,
+  msg: Buffer,
+  opts?: { ephemPrivateKey?: Buffer; iv?: Buffer } | undefined
+) {
+  opts = opts || {}
+  // Take IV from opts or generate randomly
+  const iv = opts?.iv || randomBytes(16)
+  let ephemPrivateKey = opts?.ephemPrivateKey || randomBytes(32)
+  // There is a very unlikely possibility that it is not a valid key
+  while (!isValidPrivateKey(ephemPrivateKey)) {
+    if (opts?.ephemPrivateKey) {
+      throw new Error('ephemPrivateKey is not valid')
+    }
+    ephemPrivateKey = randomBytes(32)
+  }
+  // Get the public key from the ephemeral private key
+  const ephemeralPublicKey = getPublic(ephemPrivateKey)
+
+  const hash = await sha512(await derive(ephemPrivateKey, publicKeyTo))
+  const encryptionKey = hash.slice(0, 32)
+  const macKey = hash.slice(32)
+  const ciphertext = await aesCbcEncrypt(iv, encryptionKey, msg)
+
+  // Get a MAC
+  const dataToMac = Buffer.concat([iv, ephemeralPublicKey, ciphertext])
+  const mac = await hmacSha256Sign(macKey, dataToMac)
+
+  // Return the payload
+  return {
+    iv,
+    ephemeralPublicKey,
+    ciphertext,
+    mac,
+  }
+}
+
+export async function decrypt(privateKey: Buffer, opts: Ecies) {
```

Comment:
Also converted to `async/await` from [original](https://github.com/bitchan/eccrypto/blob/master/browser.js#L243) for readability

Reply:
At this point you might as well ship your own open source fork of eccrypto haha